src/masternodes/loan.h

```diff
@@ -0,0 +1,252 @@
+{
+public:
+    uint256 creationTx;
+    int32_t creationHeight = -1;
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        READWRITEAS(CLoanSetCollateralToken, *this);
+        READWRITE(creationTx);
+        READWRITE(creationHeight);
+    }
+};
+
+struct CLoanSetCollateralTokenMessage : public CLoanSetCollateralToken {
+    using CLoanSetCollateralToken::CLoanSetCollateralToken;
+
+    ADD_SERIALIZE_METHODS;
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        READWRITEAS(CLoanSetCollateralToken, *this);
+    }
+};
+
+class CLoanSetLoanToken
+{
+public:
+    std::string symbol;
+    std::string name;
+    uint256 priceFeedTxid;
+    bool mintable = false;
+    CAmount interest = 0;
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        READWRITE(symbol);
+        READWRITE(name);
+        READWRITE(priceFeedTxid);
+        READWRITE(mintable);
+        READWRITE(interest);
+    }
+};
+
+class CLoanSetLoanTokenImplementation : public CLoanSetLoanToken
+{
+public:
+    uint256 creationTx;
+    int32_t creationHeight = -1;
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        READWRITEAS(CLoanSetLoanToken, *this);
+        READWRITE(creationTx);
+        READWRITE(creationHeight);
+    }
+};
+
+struct CLoanSetLoanTokenMessage : public CLoanSetLoanToken {
+    using CLoanSetLoanToken::CLoanSetLoanToken;
+
+    ADD_SERIALIZE_METHODS;
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        READWRITEAS(CLoanSetLoanToken, *this);
+    }
+};
+
+struct CLoanUpdateLoanTokenMessage : public CLoanSetLoanToken {
+    using CLoanSetLoanToken::CLoanSetLoanToken;
+
+    uint256 tokenTx;
+
+    ADD_SERIALIZE_METHODS;
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        READWRITEAS(CLoanSetLoanToken, *this);
+        READWRITE(tokenTx);
+    }
+};
+
+struct CLoanSchemeData
+{
+    uint32_t ratio;
+    CAmount rate;
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        READWRITE(ratio);
+        READWRITE(rate);
+    }
+};
+
+struct CLoanScheme : public CLoanSchemeData
+{
+    std::string identifier;
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        READWRITEAS(CLoanSchemeData,*this);
+        READWRITE(identifier);
+    }
+};
+
+struct CLoanSchemeMessage : public CLoanScheme
+{
+    uint64_t update{0};
```

Comment:
The variable name is quite misleading, call something like `updateHeight` maybe better.